### ISSUE TITLE
before_all hooks

### ISSUE THREAD

**Issue by danielwaterworth** [palkan/test-prof#133]
I'm working on speeding up the [discourse](https://github.com/discourse/discourse) tests using `let_it_be`. We need a way to run code that sets up global variables before each before_all block is invoked. I can do this by providing a custom `BeforeAll` adapter within discourse, but if you would like, I could make it more general and submit a PR here. What would you prefer?

**Comment by palkan** [comment]
Hey @danielwaterworth!

> We need a way to run code that sets up global variables before each before_all block is invoked

Yeah, I see (https://github.com/discourse/discourse/pull/7414/files#diff-e11bdaf9116be3911e9557b69e242370R213).

>  I could make it more general and submit a PR here

Yeah, that would be great.
We already have an example where it could be useful: https://github.com/palkan/test-prof/blob/a783083988e6b461f7ad32aa891ad57962a1b97d/lib/test_prof/before_all/adapters/active_record.rb#L15

It could have the following API:

```ruby
Before.configure do |config|
  config.setup_before_all { }
  config.teardown_before_all { }
end
```

**Comment by palkan** [comment]
Closed by #140
